src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,44 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { TodoStatusFilter } from '../../constants/StatusFilter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const onQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(filterActions.setQuery(event.target.value));
+  };
+
+  const onQueryClear = () => {
+    dispatch(filterActions.clearQuery());
+  };
+
+  const onStatusChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterActions.setStatus(event.target.value as TodoStatusFilter));
+  };
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={onStatusChange}
+          >
+            {Object.keys(TodoStatusFilter).map((statusFilter, index) => (
+              <option
+                key={`${statusFilter}-${index}`}
+                value={statusFilter.toLowerCase()}
```

Comment:
Ensure that the `value` attribute in the `<option>` elements matches the expected values in the Redux state. If the state values are not stored in lowercase, this may cause issues with filtering.